LadybugTools_Engine/Python/src/ladybugtools_toolkit/solar.py

```diff
@@ -952,3 +914,337 @@ def plot_directional_irradiance(
+                        anchor = val - offset_distance
+                    ax.text(
+                        theta,
+                        anchor,
+                        f"{val:0.0f}",
+                        fontsize="xx-small",
+                        ha=ha,
+                        va="center",
+                        rotation=-theta_deg - 90,
+                        rotation_mode="anchor",
+                        color=contrasting_color(color),
+                    )
+
+    ax.set_title(
+        f"{epw_file.name}\n{rad_type.title()} irradiance ({tilt_angle}° altitude)\n{describe_analysis_period(analysis_period)}"
+    )
+
+    plt.tight_layout()
+
+    return ax
+
+
+def tilt_orientation_factor(
+    epw_file: Path,
+    ax: plt.Axes = None,
+    rad_type: str = "total",
+    analysis_period: AnalysisPeriod = AnalysisPeriod(),
+    cmap: str = "YlOrRd",
+    directions: int = 36,
+    tilts: int = 9,
+    quantiles: tuple[float] = (0.05, 0.25, 0.5, 0.75, 0.95),
+    lims: tuple[float, float] = None,
+) -> plt.Axes:
+    """Create a tilt-orientation factor plot.
+
+    Args:
+        epw_file (Path):
+            The EPW file representing the weather data/location to be visualised.
+        ax (plt.Axes, optional):
+            The axes to plot on.
+            Defaults to None.
+        rad_type (str, optional):
+            The type of radiation to plot.
+            Defaults to "total", with options of "total", "direct" and "diffuse".
+        analysis_period (AnalysisPeriod, optional):
+            The analysis period over which radiation shall be summarised.
+            Defaults to AnalysisPeriod().
+        cmap (str, optional):
+            The colormap to apply.
+            Defaults to "YlOrRd".
+        directions (int, optional):
+            The number of directions to bin data into.
+            Defaults to 36.
+        tilts (int, optional):
+            The number of tilts to calculate.
+            Defaults to 9.
+        quantiles (tuple[float], optional):
+            The quantiles to plot.
+            Defaults to (0.05, 0.25, 0.5, 0.75, 0.95).
+        lims (tuple[float, float], optional):
+            The limits of the plot.
+            Defaults to None.
+
+    Returns:
+        plt.Axes:
+            The matplotlib axes.
+    """
+
+    # create dir for cached results
+    _dir = Path(hb_folders.default_simulation_folder) / "_lbt_tk_solar"
+    _dir.mkdir(exist_ok=True, parents=True)
+    ndir = directions
+
+    if ax is None:
+        ax = plt.gca()
+
+    cmap = plt.get_cmap(cmap)
+
+    # create sky matrix
+    smx = SkyMatrix.from_epw(epw_file=epw_file, high_density=True, hoys=analysis_period.hoys)
+
+    # create roses per tilt angle
+    _directions = np.linspace(0, 360, directions + 1)[:-1].tolist()
+    _tilts = np.linspace(0, 90, tilts)[:-1].tolist() + [89.999]
+    rrs: list[RadiationRose] = []
+    for ta in tqdm(_tilts):
+        sp = _dir / f"{epw_file.stem}_{ndir}_{ta:0.4f}.pickle"
```

Comment:
This path also needs to include the analysis period description, as the analysis period is taken into account when creating the sky matrix.
Or some sort of representation of `analysis_period.hoys`, though I think ap start and end dates/times would be best.

As it currently is, if using the same tilts, directions and epw file, but different analysis periods (eg. comparing summer and winter), then one analysis period would be pickled, and then future analysis periods would only show the first used, ignoring the rest.

Reply:
Is this relevant? The method doesn't iterate over multiple analysis periods.

Reply:
@jamesramsden-bh It is relevant - while the method itself doesn't iterate over multiple analysis periods, if you wished to create plots for multiple analysis periods while keeping all other parameters the same, this would save the results of the first analysis period you simulate, then use those saved results for all the other analysis periods - in my example you would get a summer angle for winter months (despite the analysis period). You can see this for yourself if you run the method with the following:

```python
aps = [AnalysisPeriod(st_month=1, end_month=4), AnalysisPeriod(st_month=6, end_month=10)]

axs = []

for ap in aps:
    axs.append(tile_orientation_factor(epw, analysis_period=ap))

```
(assuming you have already imported everything and initialised an epw object)